pandas/tests/frame/methods/test_dtypes.py

```diff
@@ -79,6 +80,21 @@ def test_dtypes_are_correct_after_column_slice(self):
             Series({"a": np.float_, "b": np.float_, "c": np.float_}),
         )
 
+    @pytest.mark.parametrize(
+        "data",
+        [
+            DataFrame({"id": [1, 2, 3, 4], "test": [True, pd.NA, pd.NA, False]}),
+            DataFrame({"id": [1, 2, 3, 4], "test": [True, pd.NA, True, False]}),
```

Comment:
Everything that is common can be moved down into the function too. i.e

```
"data": [pd.NA, True])
def test...(self, data)
    data = DataFrame({"id": [1, 2, 3, 4], "test": [True, pd.NA, data, False]})
```

Reply:
I have now moved everything down except the test arrays.